src/components/filter/filter.tsx

```diff
@@ -184,7 +185,7 @@ export class Filter
   async setFocus(): Promise<void> {
     await componentLoaded(this);
 
-    focusElement(this.textInput);
+    this.el?.focus();
```

Comment:
Wouldn't this no longer focus the textInput?

Reply:
Yeah the Input is the first focusable element. I double checked to be sure, putting this in the demos for this branch correctly focuses the Input:

```html
  <calcite-filter></calcite-filter>
  <script>
    customElements.whenDefined("calcite-filter").then(() => 
        document.querySelector("calcite-filter").setFocus());
  </script>
```

> If we call `focus()` on a component with delegatesFocus enabled does it automatically focus the first element?



It depends on whether everything is loaded. For example this works, but not without the timeout:

```js
    customElements
      .whenDefined("calcite-filter")
      .then(() => setTimeout(() => document.querySelector("calcite-filter").focus(), 1000));
```

We still need the async `setFocus()` methods.